src/node/blockstorage.cpp

```diff
@@ -199,18 +196,13 @@ CBlockIndex* BlockManager::InsertBlockIndex(const uint256& hash)
         return nullptr;
     }
 
-    // Return existing
-    BlockMap::iterator mi = m_block_index.find(hash);
-    if (mi != m_block_index.end()) {
-        return (*mi).second;
+    // Return existing or create new
+    auto [mi, inserted] = m_block_index.try_emplace(hash);
```

Comment:
```suggestion
    const auto [mi, inserted]{m_block_index.try_emplace(hash)};
```

In the third commit: I think try_emplace is self-explanatory and the comment can be removed